最新版代码densenet_161_gpu 8P graph 模型训练报错

If this is your first time, please read our contributor guidelines:
https://github.com/mindspore-lab/mindcv/blob/main/CONTRIBUTING.md

**Describe the bug/ 问题描述 (Mandatory / 必填)**
最新版代码densenet_161_gpu 8P graph 模型训练报错

- **Hardware Environment(`Ascend`/`GPU`/`CPU`)  / 硬件环境**:
> Please delete the backend not involved / 请删除不涉及的后端:
> /device GPU

- **Software Environment / 软件环境 (Mandatory / 必填)**:
-- MindSpore version (e.g., 1.7.0.Bxxx) :__commit_id__ = ''[sha1]:c69cc34d,[branch]:(HEAD,origin/r1.8,r1.8)''     Version: 1.8.1
-- Python version (e.g., Python 3.7.5) :Python 3.7.5
-- OS platform and distribution (e.g., Linux Ubuntu 16.04): Linux  Ubuntu  18.04
-- GCC/Compiler version (if compiled from source):gcc version 7.3.0 (GCC)

- **Excute Mode / 执行模式 (Mandatory / 必填)(`PyNative`/`Graph`)**:
> Please delete the mode not involved / 请删除不涉及的模式:
> /mode graph

**To Reproduce / 重现步骤 (Mandatory / 必填)**
Steps to reproduce the behavior:、
1. 拉取3月16号  main分支源码
2. 用该源码执行  mpirun --allow-run-as-root -n 8 python train.py -c configs/densenet/densenet_161_gpu.yaml 

**Expected behavior / 预期结果 (Mandatory / 必填)**
mpirun --allow-run-as-root -n 8 python train.py -c configs/densenet/densenet_161_gpu.yaml  正常训练

**Screenshots/ 日志 / 截图 (Mandatory / 必填)**
[ModelZoo-compile_time:0:00:00.006793]
[ModelZoo-compile_time:0:00:12.142973]
[ModelZoo-compile_time:0:00:00.006730]
[ModelZoo-compile_time:0:00:12.153091]
[ModelZoo-compile_time:0:00:00.008439]
[ModelZoo-compile_time:0:00:00.006708]
[ModelZoo-compile_time:0:00:12.157156]
[ModelZoo-compile_time:0:00:12.150117]
[ModelZoo-compile_time:0:00:00.006696]
[ModelZoo-compile_time:0:00:12.154727]
[ModelZoo-compile_time:0:00:00.006641]
[ModelZoo-compile_time:0:00:12.163353]
[ModelZoo-compile_time:0:00:00.006710]
[ModelZoo-compile_time:0:00:12.161039]
[ModelZoo-compile_time:0:00:00.008573]
[ModelZoo-compile_time:0:00:12.172610]
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
[WARNING]: warmup_epochs + decay_epochs > num_epochs. Please check and reduce decay_epochs!
----------------------------------------
Num devices: 8
Distributed mode: True
Num training samples: 1281168
Num classes: 1000
Num batches: 5004
Batch size: 32
Auto augment: None
Model: densenet161
Model param: 28900936
Num epochs: 2
Optimizer: momentum
LR: 0.1
LR Scheduler: cosine_decay
----------------------------------------
Start training
[ModelZoo-compile_time:0:00:27.475201]
[ModelZoo-compile_time:0:00:26.705203]
[ModelZoo-compile_time:0:00:27.297047]
[ModelZoo-compile_time:0:00:29.536742]
[ModelZoo-compile_time:0:00:28.118034]
[ModelZoo-compile_time:0:00:27.606911]
[ModelZoo-compile_time:0:00:31.039859]
[ModelZoo-compile_time:0:00:28.869868]
[ModelZoo-compile_time:0:32:57.310788]
[ModelZoo-compile_time:0:32:54.162378]
[ModelZoo-compile_time:0:32:53.445363]
[ModelZoo-compile_time:0:32:55.635937]
[ModelZoo-compile_time:0:32:55.648902]
[ModelZoo-compile_time:0:32:54.478779]
[ModelZoo-compile_time:0:32:58.475726]
[ModelZoo-compile_time:0:32:57.024846]
[ModelZoo-compile_time:0:00:00.023840]
[ModelZoo-compile_time:0:00:00.024558]
[ModelZoo-compile_time:0:00:00.032221]
[ModelZoo-compile_time:0:00:00.036640]
[ModelZoo-compile_time:0:00:00.038472]
[ModelZoo-compile_time:0:00:00.037105]
[ModelZoo-compile_time:0:00:00.040945]
[ModelZoo-compile_time:0:00:00.039485]
[ModelZoo-compile_time:0:00:01.046918]
[ModelZoo-compile_time:0:00:01.034783]
[ModelZoo-compile_time:0:00:01.043079]
[ModelZoo-compile_time:0:00:01.026836]
[ModelZoo-compile_time:0:00:01.043748]
[ModelZoo-compile_time:0:00:01.020719]
[ModelZoo-compile_time:0:00:01.036203]
[ModelZoo-compile_time:0:00:01.046423]
Epoch: 1, batch:[5004/5004], loss:4.495938, lr: 0.0999829,  time:3502.724194s
Epoch: 1, batch:[5004/5004], loss:4.928809, lr: 0.0999829,  time:3502.957811s
Epoch: 1, batch:[5004/5004], loss:4.951904, lr: 0.0999829,  time:3502.695719s
Epoch: 1, batch:[5004/5004], loss:4.591516, lr: 0.0999829,  time:3502.947372s
Epoch: 1, batch:[5004/5004], loss:4.765549, lr: 0.0999829,  time:3502.677697s
Epoch: 1, batch:[5004/5004], loss:4.857996, lr: 0.0999829,  time:3502.931524s
Epoch: 1, batch:[5004/5004], loss:4.694203, lr: 0.0999829,  time:3502.802843s
Epoch: 1, batch:[5004/5004], loss:5.142710, lr: 0.0999829,  time:3502.942353s
[ModelZoo-compile_time:0:00:00.000128]
[ModelZoo-compile_time:0:00:00.000126]
[ModelZoo-compile_time:0:00:00.000131]
[ModelZoo-compile_time:0:00:00.000124]
[ModelZoo-compile_time:0:00:00.000141]
[ModelZoo-compile_time:0:00:00.000154]
[ModelZoo-compile_time:0:00:00.000182]
[ERROR] RUNTIME_FRAMEWORK(13171,7f458fffb700,python):2023-03-16-12:13:22.365.803 [mindspore/ccsrc/runtime/graph_scheduler/actor/abstract_actor.cc:99] EraseInput] Erase input data failed: Default/Add-op12401, sequential_num: 1226346840
[ERROR] RUNTIME_FRAMEWORK(13171,7f45917fe700,python):2023-03-16-12:13:22.365.807 [mindspore/ccsrc/runtime/graph_scheduler/actor/abstract_actor.cc:99] EraseInput] Erase input data failed: Gradients/Default/network-TrainOneStepCell/network-WithLossCell/_backbone-DenseNet/features-SequentialCell/transition1-_Transition/features-SequentialCell/conv-Conv2d/gradConv2D/Conv2DBackpropInput-op12477, sequential_num: 1226346840
[WARNING] MD(13169,7fdf4fc12740,python):2023-03-16-12:13:22.712.537 [mindspore/ccsrc/minddata/dataset/engine/datasetops/device_queue_op.cc:75] ~DeviceQueueOp] preprocess_batch: 5008; batch_queue: 15, 16, 14, 16, 15, 16, 15, 16, 16, 16; push_start_time: 2023-03-16-12:13:18.246.232, 2023-03-16-12:13:18.539.609, 2023-03-16-12:13:18.840.954, 2023-03-16-12:13:19.141.508, 2023-03-16-12:13:19.454.234, 2023-03-16-12:13:19.764.165, 2023-03-16-12:13:20.079.815, 2023-03-16-12:13:20.392.762, 2023-03-16-12:13:20.713.789, 2023-03-16-12:13:22.111.227; push_end_time: 2023-03-16-12:13:18.539.585, 2023-03-16-12:13:18.840.931, 2023-03-16-12:13:19.130.670, 2023-03-16-12:13:19.449.198, 2023-03-16-12:13:19.764.139, 2023-03-16-12:13:20.079.767, 2023-03-16-12:13:20.392.722, 2023-03-16-12:13:20.713.764, 2023-03-16-12:13:22.110.501, 2023-03-16-12:13:22.703.120.
[WARNING] MD(13171,7f474fab6740,python):2023-03-16-12:13:22.759.154 [mindspore/ccsrc/minddata/dataset/engine/datasetops/device_queue_op.cc:75] ~DeviceQueueOp] preprocess_batch: 5008; batch_queue: 15, 16, 15, 16, 15, 16, 15, 16, 16, 16; push_start_time: 2023-03-16-12:13:18.246.389, 2023-03-16-12:13:18.539.525, 2023-03-16-12:13:18.840.888, 2023-03-16-12:13:19.130.594, 2023-03-16-12:13:19.439.598, 2023-03-16-12:13:19.769.929, 2023-03-16-12:13:20.079.746, 2023-03-16-12:13:20.419.320, 2023-03-16-12:13:20.713.923, 2023-03-16-12:13:22.114.132; push_end_time: 2023-03-16-12:13:18.539.489, 2023-03-16-12:13:18.840.851, 2023-03-16-12:13:19.130.571, 2023-03-16-12:13:19.439.565, 2023-03-16-12:13:19.769.866, 2023-03-16-12:13:20.079.711, 2023-03-16-12:13:20.419.269, 2023-03-16-12:13:20.713.887, 2023-03-16-12:13:22.112.706, 2023-03-16-12:13:22.757.637.
[WARNING] MD(13175,7facdd80c740,python):2023-03-16-12:13:22.796.228 [mindspore/ccsrc/minddata/dataset/engine/datasetops/device_queue_op.cc:75] ~DeviceQueueOp] preprocess_batch: 5008; batch_queue: 15, 16, 15, 16, 15, 16, 15, 16, 15, 16; push_start_time: 2023-03-16-12:13:18.246.662, 2023-03-16-12:13:18.539.626, 2023-03-16-12:13:18.840.976, 2023-03-16-12:13:19.130.441, 2023-03-16-12:13:19.439.504, 2023-03-16-12:13:19.764.160, 2023-03-16-12:13:20.079.598, 2023-03-16-12:13:20.393.423, 2023-03-16-12:13:20.713.565, 2023-03-16-12:13:22.141.081; push_end_time: 2023-03-16-12:13:18.539.604, 2023-03-16-12:13:18.840.951, 2023-03-16-12:13:19.130.399, 2023-03-16-12:13:19.439.481, 2023-03-16-12:13:19.764.120, 2023-03-16-12:13:20.079.539, 2023-03-16-12:13:20.393.400, 2023-03-16-12:13:20.713.528, 2023-03-16-12:13:22.141.026, 2023-03-16-12:13:22.794.886.
Traceback (most recent call last):
  File "train.py", line 307, in <module>
    train(args)
  File "train.py", line 293, in train
    trainer.train(args.epoch_size, loader_train, callbacks=callbacks, dataset_sink_mode=args.dataset_sink_mode)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 1049, in train
    initial_epoch=initial_epoch)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 98, in wrapper
    func(self, *args, **kwargs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 623, in _train
    cb_params, sink_size, initial_epoch, valid_infos)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 701, in _train_dataset_sink_process
    outputs = train_network(*inputs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/cell.py", line 578, in __call__
    out = self.compile_and_run(*args)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/cell.py", line 998, in compile_and_run
    return _cell_graph_executor(self, *new_inputs, phase=self.phase)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1202, in __call__
    return self.run(obj, *args, phase=phase)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1239, in run
    return self._exec_pip(obj, *args, phase=phase_real)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 98, in wrapper
    results = fn(*arg, **kwargs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1221, in _exec_pip
    return self._graph_executor(args, phase)
RuntimeError: cuDNN Error: ConvolutionBackwardFilter failed | Error Number: 3 CUDNN_STATUS_BAD_PARAM
The function call stack:
Corresponding code candidate:
 - In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/ops/_grad/grad_nn_ops.py(107)/        dw = filter_grad(dout, x, w_shape)/
Corresponding forward node candidate:
 - In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/layer/conv.py(287)/        output = self.conv2d(x, self.weight)/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(67)/        bottleneck = self.conv1(self.relu1(self.norm1(features)))/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(98)/        for layer in self.cell_list:/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/layer/container.py(275)/        for cell in self.cell_list:/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(210)/        x = self.features(x)/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(219)/        x = self.forward_features(x)/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/amp.py(222)/            out = self._backbone(data)/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/wrap/cell_wrapper.py(376)/        loss = self.network(*inputs)/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/dataset_helper.py(106)/        return self.network(*outputs)/

----------------------------------------------------
- C++ Call Stack: (For framework developers)
----------------------------------------------------
mindspore/ccsrc/plugin/device/gpu/kernel/nn/conv2d_grad_filter_gpu_kernel.h:118 Launch

Traceback (most recent call last):
  File "train.py", line 307, in <module>
    train(args)
  File "train.py", line 293, in train
    trainer.train(args.epoch_size, loader_train, callbacks=callbacks, dataset_sink_mode=args.dataset_sink_mode)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 1049, in train
    initial_epoch=initial_epoch)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 98, in wrapper
    func(self, *args, **kwargs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 623, in _train
    cb_params, sink_size, initial_epoch, valid_infos)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 701, in _train_dataset_sink_process
    outputs = train_network(*inputs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/cell.py", line 578, in __call__
    out = self.compile_and_run(*args)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/cell.py", line 998, in compile_and_run
    return _cell_graph_executor(self, *new_inputs, phase=self.phase)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1202, in __call__
    return self.run(obj, *args, phase=phase)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1239, in run
    return self._exec_pip(obj, *args, phase=phase_real)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 98, in wrapper
    results = fn(*arg, **kwargs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1221, in _exec_pip
    return self._graph_executor(args, phase)
RuntimeError: cuDNN Error: cudnnConvolutionForward failed | Error Number: 4 CUDNN_STATUS_INTERNAL_ERROR
The function call stack:
In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/layer/conv.py(287)/        output = self.conv2d(x, self.weight)/
In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(68)/        new_features = self.conv2(self.relu2(self.norm2(bottleneck)))/
In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(98)/        for layer in self.cell_list:/
In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/layer/container.py(275)/        for cell in self.cell_list:/
In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(210)/        x = self.features(x)/
In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(219)/        x = self.forward_features(x)/
In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/amp.py(222)/            out = self._backbone(data)/
In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/wrap/cell_wrapper.py(376)/        loss = self.network(*inputs)/
In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/dataset_helper.py(106)/        return self.network(*outputs)/

----------------------------------------------------
- C++ Call Stack: (For framework developers)
----------------------------------------------------
mindspore/ccsrc/plugin/device/gpu/kernel/nn/conv2d_gpu_kernel.h:71 Launch

Saving model to ./ckpt/densenet161-1_5004.ckpt
Total time since last epoch: 3505.033
--------------------------------------------------------------------------------
[ModelZoo-compile_time:0:00:00.000149]
Traceback (most recent call last):
  File "train.py", line 307, in <module>
    train(args)
  File "train.py", line 293, in train
    trainer.train(args.epoch_size, loader_train, callbacks=callbacks, dataset_sink_mode=args.dataset_sink_mode)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 1049, in train
    initial_epoch=initial_epoch)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 98, in wrapper
    func(self, *args, **kwargs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 623, in _train
    cb_params, sink_size, initial_epoch, valid_infos)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/model.py", line 701, in _train_dataset_sink_process
    outputs = train_network(*inputs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/cell.py", line 578, in __call__
    out = self.compile_and_run(*args)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/cell.py", line 998, in compile_and_run
    return _cell_graph_executor(self, *new_inputs, phase=self.phase)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1202, in __call__
    return self.run(obj, *args, phase=phase)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1239, in run
    return self._exec_pip(obj, *args, phase=phase_real)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 98, in wrapper
    results = fn(*arg, **kwargs)
  File "/root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/common/api.py", line 1221, in _exec_pip
    return self._graph_executor(args, phase)
RuntimeError: cuDNN Error: ConvolutionBackwardData failed | Error Number: 3 CUDNN_STATUS_BAD_PARAM
The function call stack:
Corresponding code candidate:
 - In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/ops/_grad/grad_nn_ops.py(105)/        dx = input_grad(dout, w, x_shape)/
Corresponding forward node candidate:
 - In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/layer/conv.py(287)/        output = self.conv2d(x, self.weight)/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(68)/        new_features = self.conv2(self.relu2(self.norm2(bottleneck)))/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(98)/        for layer in self.cell_list:/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/layer/container.py(275)/        for cell in self.cell_list:/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(210)/        x = self.features(x)/
   In file /media/nvme1n1/cty/test_mindcv/latest_mindcv/mindcv/mindcv/models/densenet.py(219)/        x = self.forward_features(x)/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/amp.py(222)/            out = self._backbone(data)/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/nn/wrap/cell_wrapper.py(376)/        loss = self.network(*inputs)/
   In file /root/miniconda3/envs/mind_lab/lib/python3.7/site-packages/mindspore/train/dataset_helper.py(106)/        return self.network(*outputs)/

----------------------------------------------------
- C++ Call Stack: (For framework developers)
----------------------------------------------------
mindspore/ccsrc/plugin/device/gpu/kernel/nn/conv2d_grad_input_gpu_kernel.h:117 Launch





**Additional context / 备注 (Optional / 选填)**

详细日志如上，因为以前的mindcv源码是可以正常训练的，并且通过多种组合方案，已经可以缩小报错原因是今天的最新源码导致，按照复现流程去解单即可

--- a/train.py
+++ b/train.py
@@ -36,7 +36,7 @@ def train(args):
             device_num=device_num,
             parallel_mode="data_parallel",
             gradients_mean=True,
-            parameter_broadcast=True,
+            # we should but cannot set parameter_broadcast=True, which will cause error on gpu.
         )
     else:
         device_num = None

